winml/lib/Api.Core/WinMLAdapter.cpp

```diff
@@ -180,4 +180,16 @@ onnxruntime::Tensor* STDMETHODCALLTYPE CreateTensor(
   return pTensor;
 }
 
+WinML::LotusEnvironment* STDMETHODCALLTYPE CreateLotusEnvironment()
+{
+  return PheonixSingleton<WinML::LotusEnvironment>().get();
```

Comment:
>PheonixSingleton<WinML::LotusEnvironment>().get(); [](start = 9, length = 50)

so this is going to do goofy things.    the PheonixSingleton() function returns a shared_ptr on the stack.   the call to .get() gets the pointer.   then when CreateLotusEnvironment() returns it will destroy the shared_ptr from the stack.   destroying the pheonix.  right ?

Reply:
Yeah that's true. Let me think about it some more